sycl/include/sycl/detail/pi.h

```diff
@@ -127,6 +131,11 @@ using pi_bool = pi_uint32;
 using pi_bitfield = pi_uint64;
 using pi_native_handle = uintptr_t;
 
+typedef struct _ze_command_queue_handle_t *ze_command_queue_handle_t;
+typedef struct _ze_command_list_handle_t *ze_command_list_handle_t;
+using pi_native_handle2 =
+    std::variant<ze_command_queue_handle_t, ze_command_list_handle_t>;
```

Comment:
```suggestion
using pi_native_handle_variant =
    std::variant<pi_native_handle, pi_native_handle>;
```

Reply:
Unfortunately, this doesn't work. There are cascading type mismatch errors that result from this change.

Reply:
We must find a way to not expose L0 types in the pi.h

Reply:
I have moved the pi_native_handle2 definition to queue.hpp.